src/pages/discord/link/index.page.tsx

```diff
@@ -0,0 +1,42 @@
+import { useWallet } from "@solana/wallet-adapter-react";
+import { useEffect, useState } from "react";
+import { useCluster } from "~/components/ClusterProvider";
+import { Redirect } from "~/components/common/Redirect";
+import { getDiscordUser } from "~/network/discord";
+import { usePlayerStore } from "~/stores/usePlayerStore";
+import { getRoute } from "~/utils/getRoute";
+import Card from "./components/card";
+
+const DiscordLink = () => {
+  const { publicKey } = useWallet();
+  const self = usePlayerStore((state) => state.self);
```

Comment:
As we use self here, we can use the `SelfRetriever` component and do this stuff inside the `View` component